### ISSUE TITLE
CommonJS or AMD dependencies can cause optimization bailouts in Angular 10

### ISSUE THREAD

**Issue by zijianhuang** [mattlewis92/angular-calendar#1292]
**Describe the bug**

When building my Angular app with Angular 10, I see the following warning:

```
WARNING in i\NGSource\node_modules\@mattlewis92\dom-autoscroller\dist\bundle.js depends on dom-plane. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

WARNING in \NGSource\node_modules\@mattlewis92\dom-autoscroller\dist\bundle.js depends on dom-set. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies
```

This is described at https://blog.angular.io/version-10-of-angular-now-available-78960babd41

**Minimal reproduction of the problem with instructions**

Any Angular 10 project with the calendar component.

**Screenshots**

n/a

**Versions**

```js
  "dependencies": {
    "@angular/animations": "^10.0.2",
    "@angular/cdk": "10.0.1",
    "@angular/common": "10.0.2",
    "@angular/compiler": "^10.0.2",
    "@angular/core": "^10.0.2",
    "@angular/flex-layout": "^10.0.0-beta.32",
    "@angular/forms": "^10.0.2",
    "@angular/localize": "^10.0.2",
    "@angular/material": "10.0.1",
    "@angular/material-moment-adapter": "10.0.1",
    "@angular/platform-browser": "^10.0.2",
    "@angular/platform-browser-dynamic": "^10.0.2",
    "@angular/platform-server": "^10.0.2",
    "@angular/pwa": "0.901.7",
    "@angular/router": "^10.0.2",
    "@angular/service-worker": "^10.0.2",
    "@azure/storage-blob": "^10.3.0",
    "@azure/storage-file-share": "12.1.1",
    "@kolkov/angular-editor": "1.1.2",
    "@microsoft/signalr": "3.1.4",
    "@ng-bootstrap/ng-bootstrap": "6.1.0",
    "angular-calendar": "^0.28.16",
```

**Comment by wrappixel** [comment]
For a temporary solution, you can add this angular calendar in "allowedCommonJsDependencies": ["angular-calendar"] at the angular.json file. it resolves the warning.

**Comment by m-sadegh-sh** [comment]
> For a temporary solution, you can add this angular calendar in "allowedCommonJsDependencies": ["angular-calendar"] at the angular.json file. it resolves the warning.

Eliminating warning doesn't provide any feedback, It would be awesome to fix the actual problem. 🤦‍♂️

**Comment by mattlewis92** [comment]
Hey folks, thanks for your patience on this, I've been a little bit busy the last few months but finally had the time this weekend to put out a fix. If you update to `0.28.17` then it should fix the warning message and your bundle sizes should be a little bit smaller as a result 😄

**Comment by Sjeiti** [comment]
I'm on `0.28.22` and getting `WARNING in (...)\node_modules\angular-calendar\date-adapters\esm\date-fns\index.js depends on 'date-fns'. CommonJS or AMD dependencies can cause optimization bailouts.`

Different warnings than above, but same category?
Or will this be addressed by #1417 ?
